x-pack/plugins/ingest_manager/public/applications/ingest_manager/hooks/use_intra_app_state.tsx

```diff
@@ -28,7 +28,7 @@ export const IntraAppStateProvider = memo<{
 }>(({ kibanaScopedHistory, children }) => {
   const internalAppToAppState = useMemo<IntraAppState>(() => {
     return {
-      forRoute: kibanaScopedHistory.location.hash.substr(1),
+      forRoute: kibanaScopedHistory.location.hash.substr(1).split('?')[0],
```

Comment:
It's not clear to me what's happening here, and I'm a bit uneasy with the chain of string operations on the url.

After playing around a bit I think I see the intent and, IIUC, I'd prefer something more explicit/safe like `URL` which is available in the browser and node
```typescript
const route = kibanaScopedHistory.location.hash.substr(1);
const url = new URL(route, kibanaScopedHistory.location);
...
    forRoute: url.pathname
```

Reply:
@jfsiii @paul-tavares I solved this by just passing in the url base path as well from the state so that we can control it from there.  It saves us from having to parse potential query params and is more exact.  let me know your thoughts